service/src/stage_service.rs

```diff
@@ -139,8 +159,9 @@ impl StageService for StageServiceSVC {
             let prove_task = stage.get_prove_task();
             if let Some(prove_task) = prove_task {
                 let tx = tx.clone();
+                let tls_config = self.tls_config.clone();
```

Comment:
It's better to implement Borrow trait TlsConfig,  since I saw many times `clone` for it.  take a look at https://stackoverflow.com/questions/22282117/how-do-i-borrow-a-reference-to-what-is-inside-an-optiont.

Reply:
It may seems difficult to completely achieve this goal(if you want to make the code clearly). And here is the reason.
1. We clone the tls_config because we are going to use it in a spawn task, which means borrow will cause lifetime problem. And there is no spawn problem in the stackoverflow example. If we want to solve the lifetime problem, clone the self.tls_config and move the variable into the spawn(or use the arc).
2. We still need to clone the config content(include ca_cert and identity) when we try to build endpoint of grpc client because it need the ownership of them, so we still need to  pay the cost of clone. But the cost may be a little small because we can reduce a clone in a for loop.
So, I implement these codes in two ways(borrow and arc, and arc will reduce three times of clone while comparing to borrow), but the codes may be more complicated😧
You can review it on branch https://github.com/evan95s/zkm-prover/tree/feature/tls-support-arc and https://github.com/evan95s/zkm-prover/tree/feature/tls-support-reference.

Reply:
Okay, this is an optimization, But keeping the code simple and clear is more important. just keep your opt code, let it opt it in the next PR.